app.py

```diff
@@ -168,35 +155,33 @@ def create_project_webhook(project_key: str, webhook_url: str, events: list[str]
         },
     }
     try:
-        response = requests.post(
+        response = await client.post(
             f"{BITBUCKET_API_URL}/rest/api/1.0/projects/{project_key}/webhooks",
             json=webhook_data,
             auth=bitbucket_auth,
         )
         response.raise_for_status()
         logger.info(f"Successfully created webhook for project {project_key}")
         return response.json()
-    except requests.exceptions.HTTPError as e:
+    except httpx.HTTPStatusError as e:
         logger.error(
             f"HTTP error when creating webhook for project {project_key}: {e.response.status_code}"
-        )
+            )
         return None
 
-
-def add_entity_to_port(blueprint_id, entity_object):
-    response = requests.post(
+async def add_entity_to_port(blueprint_id, entity_object):
+    response = await client.post(
         f"{PORT_API_URL}/blueprints/{blueprint_id}/entities?upsert=true&merge=true",
         json=entity_object,
         headers=port_headers,
     )
     logger.info(response.json())
 
-
-def get_paginated_resource(
-    path: str,
-    params: dict[str, Any] = None,
-    page_size: int = 25,
-    full_response: bool = False,
+async def get_paginated_resource(
+        path: str,
+        params: dict[str, Any] = None,
+        page_size: int = 25,
+        full_response: bool = False,
 ):
     logger.info(f"Requesting data for {path}")
```

Comment:
let's add the `params` to the logs.  Let's actually move it inside the while loop so that for every paginated request, we can see the traces